web/packages/teleport/src/Notifications/Notification.tsx

```diff
@@ -116,14 +117,15 @@ export function Notification({
   const formattedDate = formatDate(notification.createdDate);
 
   function onNotificationClick(e: React.MouseEvent<HTMLElement>) {
+    markAsClicked();
     // Prevents this from being triggered when the user is just clicking away from
     // an open "mark as read/hide this notification" menu popover.
     if (e.currentTarget.contains(e.target as HTMLElement)) {
       if (content.kind === 'text') {
         setShowTextContentDialog(true);
         return;
       }
-      // TODO rudream - add notification redirect functionality
+      history.push(content.redirectRoute);
```

Comment:
When I click on a `...requested access to a resource` notification I'm redirected to a new page, but the notifications  is not closed. This looks like a bug.